compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3266,6 +3266,20 @@ public boolean hasFillerValue(BType type) {
         }
     }
 
+    public boolean isOptionalErrorType(BType type) {
+        if (type.tag == TypeTags.UNION) {
+            BUnionType unionType = (BUnionType) type;
+            for (BType t : unionType.getMemberTypes()) {
+                if (t.tag != TypeTags.ERROR && t.tag != TypeTags.NIL) {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        return type.tag == TypeTags.ERROR;
```

Comment:
If this is the case, it isn't quite an optional error right? What if we remove this from here and instead in SemanticAnalyzer.java L2188 do 

`(type.tag == TypeTags.ERROR || types.isOptionalErrorType(expExprType))`